Author: Unrud <unrud@outlook.com>
Description: fix a bug causing the game to hang, when the window is closed during gameplay
Last-Update: 2019-05-11
--- a/src/Engine.py
+++ b/src/Engine.py
@@ -39,9 +39,9 @@
     self.running = True
 
   def quit(self):
+    self.running = False
     for t in list(self.tasks + self.frameTasks):
       self.removeTask(t)
-    self.running = False
 
   def addTask(self, task, synchronized = True):
     """
--- a/src/GameTask.py
+++ b/src/GameTask.py
@@ -42,7 +42,8 @@
     self.session.broker.removeMessageHandler(self)
     self.engine.view.popAllLayers()
     self.session.close()
-    self.engine.view.pushLayer(MainMenu.MainMenu(self.engine))
+    if self.engine.running:
+      self.engine.view.pushLayer(MainMenu.MainMenu(self.engine))
     self.engine.removeTask(self)
 
   def handleSceneEntered(self, sender, sceneId, playerId):
